packages/quantic/cypress/e2e/default-1/did-you-mean/did-you-mean.cypress.ts

```diff
@@ -0,0 +1,161 @@
+    configure(options);
+    if (options.useCase === useCaseEnum.insight) {
+      InsightInterfaceExpect.isInitialized();
+      performSearch();
+    }
+  }
+
+  useCaseParamTest.forEach((param) => {
+    describe(param.label, () => {
+      describe('when a correct search query is performed', () => {
+        it('should not display the did you mean message', () => {
+          visitDidYouMean({useCase: param.useCase});
+          cy.wait(getQueryAlias(param.useCase));
+          setQuery(exampleCorrectedQuery);
+          performSearch();
+          Expect.displayDidYouMeanLabel(false);
+          Expect.displayNoResultLabel(false);
+        });
+      });
+
+      describe('when an incorrect search query is performed but results are found', () => {
+        it('should display the did you mean message', () => {
+          visitDidYouMean({useCase: param.useCase});
+          cy.wait(getQueryAlias(param.useCase));
+          setQuery(exampleOriginalQuery);
+          mockSearchWithDidYouMean(
+            param.useCase,
+            exampleOriginalQuery,
+            exampleCorrectedQuery
+          );
+
+          scope('when executing a search', () => {
+            performSearch();
+
+            Expect.displayDidYouMeanLabel(true);
+            Expect.displayApplyCorrectionButton(true);
+            Expect.applyCorrectionButtonContains(exampleCorrectedQuery);
+            Expect.displayNoResultLabel(false);
+            Expect.displayAutomaticQueryCorrectionLabel(false);
+          });
+
+          scope('when applying the correction', () => {
+            mockSearchWithResults([], param.useCase);
+            Actions.applyCorrection();
+
+            Expect.logDidYouMeanClick(exampleCorrectedQuery);
+            Expect.displayDidYouMeanLabel(false);
+            Expect.displayApplyCorrectionButton(false);
+            Expect.displayNoResultLabel(false);
+            Expect.displayAutomaticQueryCorrectionLabel(false);
+          });
+        });
+      });
+
+      describe('when an incorrect search query is performed but no results are found', () => {
+        it('should automatically correct the query', () => {
+          visitDidYouMean({useCase: param.useCase});
+          cy.wait(getQueryAlias(param.useCase));
+          setQuery(exampleOriginalQuery);
+          mockSearchWithDidYouMeanAutomaticallyCorrected(
+            param.useCase,
+            exampleOriginalQuery,
+            exampleCorrectedQuery
+          );
+
+          performSearch();
+
+          Expect.displayDidYouMeanLabel(false);
+          Expect.displayApplyCorrectionButton(false);
+          Expect.displayNoResultLabel(true);
+          Expect.noResultLabelContains(exampleOriginalQuery);
+          Expect.displayAutomaticQueryCorrectionLabel(true);
+          Expect.automaticQueryCorrectionLabelContains(exampleCorrectedQuery);
+          Expect.logDidYouMeanAutomaticCorrection();
+        });
+      });
+
+      if (param.useCase === useCaseEnum.search) {
```

Comment:
Does it make sense to split the test in a pipeline trigger spec file instead?
What do you think?

Reply:
Sure yeah this makes sense 👌